airbyte-integrations/connectors/source-postgres/src/test-integration/java/io/airbyte/integrations/io/airbyte/integration_tests/sources/AbstractPostgresSourceDatatypeTest.java

```diff
@@ -0,0 +1,526 @@
+            .build());
+
+    // bytea stores variable length binary string
+    // https://www.postgresql.org/docs/14/datatype-binary.html
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("bytea")
+            .airbyteType(JsonSchemaType.STRING)
+            .addInsertValues("null", "decode('1234', 'hex')", "'1234'", "'abcd'", "'\\xabcd'")
+            .addExpectedValues(null, "\\x1234", "\\x31323334", "\\x61626364", "\\xabcd")
+            .build());
+
+    for (final String type : Set.of("character", "char")) {
+      addDataTypeTestData(
+          TestDataHolder.builder()
+              .sourceType(type)
+              .airbyteType(JsonSchemaType.STRING)
+              .addInsertValues("'a'", "'*'", "null")
+              .addExpectedValues("a", "*", null)
+              .build());
+
+      addDataTypeTestData(
+          TestDataHolder.builder()
+              .sourceType(type)
+              .fullSourceDataType(type + "(8)")
+              .airbyteType(JsonSchemaType.STRING)
+              .addInsertValues("'{asb123}'", "'{asb12}'")
+              .addExpectedValues("{asb123}", "{asb12} ")
+              .build());
+    }
+
+    for (final String type : Set.of("varchar", "text")) {
+      addDataTypeTestData(
+          TestDataHolder.builder()
+              .sourceType(type)
+              .airbyteType(JsonSchemaType.STRING)
+              .addInsertValues("'a'", "'abc'", "'Миші йдуть на південь, не питай чому;'", "'櫻花分店'",
+                  "''", "null", "'\\xF0\\x9F\\x9A\\x80'")
+              .addExpectedValues("a", "abc", "Миші йдуть на південь, не питай чому;", "櫻花分店", "",
+                  null, "\\xF0\\x9F\\x9A\\x80")
+              .build());
+    }
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("varchar")
+            .fullSourceDataType("character varying(10)")
+            .airbyteType(JsonSchemaType.STRING)
+            .addInsertValues("'{asb123}'", "'{asb12}'")
+            .addExpectedValues("{asb123}", "{asb12}")
+            .build());
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("cidr")
+            .airbyteType(JsonSchemaType.STRING)
+            .addInsertValues("null", "'192.168.100.128/25'", "'192.168/24'", "'192.168.1'",
+                "'128.1'", "'2001:4f8:3:ba::/64'")
+            .addExpectedValues(null, "192.168.100.128/25", "192.168.0.0/24", "192.168.1.0/24",
+                "128.1.0.0/16", "2001:4f8:3:ba::/64")
+            .build());
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("circle")
+            .airbyteType(JsonSchemaType.STRING)
+            .addInsertValues("'(5,7),10'", "'(0,0),0'", "'(-10,-4),10'", "null")
+            .addExpectedValues("<(5.0,7.0),10.0>", "<(0.0,0.0),0.0>", "<(-10.0,-4.0),10.0>", null)
+            .build());
+
+    // Debezium does not handle era indicators (AD nd BC)
```

Comment:
@subodh1810  Debezium returns `'1991-02-10'`. Value stored in DB is `'1991-02-10 BC'`

Reply:
that seems pretty problematic... Is there any way for us to bypass debezium's conversion (i.e. retrieve the raw DB value)? b/c otherwise that sounds like we currently can't resolve https://github.com/airbytehq/airbyte/issues/14590, right?

(I didn't see an obvious solution since debezium just throws an actual Date object at us, but really hoping I just missed something)

Reply:
also, we should keep the other test cases uncommented (i.e. the non-BC date, and the null entry).

Not sure if the linter will accept this, but `.addInsertValues("'1999-01-08'", /* "'1991-02-10 BC'" */, "null")` seems reasonable

Reply:
added some log statements into PostgresConverter and ran this test case; apparently we're getting a Date object with no era indicator :/ Doing some research into whether there's some other interface that would give us the raw query output

![Screen Shot 2022-07-27 at 3 02 54 PM](https://user-images.githubusercontent.com/5741425/181380377-a66b0d84-8299-40dd-8a2e-1df49249fda3.png)

Reply:
saving state before I take a long weekend:
* `java.sql.Date` does actually (kind of) support eras: if you have `Date d`, then `d.before(Date.valueOf("0001-01-01"))` is basically equivalent. (note that we use year 1 CE because there's no such thing as year 0)
    * The problem is mostly that the exact date 0001-01-01 is ambiguous (leap years, etc weren't applied consistently). But that's probably fine, since we don't really need that much precision for dates around that time?

I tried running a test case with `.addInsertValues("'0001-12-31 BC'", "'0001-01-01'")`, and was able to detect the first one as a BC date using the `.before` logic. But I dont know if the other types (Timestamp, etc) have an equivalent problem/solution.